src/persistence/net.jl

```diff
@@ -0,0 +1,78 @@
+"""
+Writes a graph `g` to a file `f` in the [Pajek
+NET](http://gephi.github.io/users/supported-graph-formats/pajek-net-format/) format.
+
+Returns 1 (number of graphs written).
+"""
+function writenet(f::IO, g::SimpleGraph)
+    println(f, "*Vertices $(nv(g))")
+    # write edges
+    if is_directed(g)
+        println(f, "*Arcs")
+    else
+        println(f, "*Edges")
+    end
+    for e in edges(g)
+        println(f, "$(src(e)) $(dst(e))")
+    end
+    return 1
+end
+
+"""
+Writes a graph `g` to STDOUT in the [Pajek
+NET](http://gephi.github.io/users/supported-graph-formats/pajek-net-format/) format.
+
+Returns 1 (number of graphs written).
+"""
+writenet(g::SimpleGraph) = writenet(STDOUT, g)
+
+"""
+Writes a graph `g` to a file `fname` in the [Pajek
+NET](http://gephi.github.io/users/supported-graph-formats/pajek-net-format/) format.
+
+Returns 1 (number of graphs written).
+"""
+function writenet(fname::AbstractString, g::SimpleGraph)
+    f = open(fname, "w")
+     writenet(f, g)
+     close(f)
```

Comment:
this should be in a do block or a try, catch, finally so that a failed write still closes the file https://julia.readthedocs.org/en/release-0.4/manual/networking-and-streams/?highlight=file#working-with-files explains the do block for this use case.